File: bin/cmds/ssh.pl
#!/usr/bin/env perl

use strict;
use warnings;

my $port=shift(@ARGV);
my $userhost=shift(@ARGV);
my $password=shift(@ARGV);
my $cmd=join(" ",@ARGV);

open(P,"ssh.exp $port $userhost $password '$cmd' |") or die;
my $l=0;
my $pw=0;
my $s="";
while (<P>) {
	s/\r//g;
	++$l;
	if (($l==1) && /^spawn/) {
		$s.=$_;
		next;
	}
	if ($pw==0 && /assword:/) {
		$pw=1;
		$s.=$_;
		next;
	}
	if ($pw==0) {
		$s.=$_;
		next;
	}
	if (/\+\+\+\-exit\-\+\+\+ (\d+)/) {
		print STDERR "$s\n";
		print STDERR "exit $1\n";
		exit $1;
	}
	print;
}
if (!close(P)) {
	print STDERR "$s";
	exit 2;
}

File: bin/cmds/ssh.py
#!/usr/bin/env python

import os
import sys
import re

def main():

    args = " "
    for s in sys.argv[1:]:
        args += ('\'' + s + '\' ')

    pw = False
    l = 0
    s = ""
    exitcode = 0

    r = os.popen("ssh.exp" + args)
    lines = r.readlines()
    for line in lines:
        l += 1
        line = line.replace('\r\n', '\n')
        if l == 1 and re.search('^spawn', line):
            s += line
            continue

        if pw is False and re.search('assword:', line):
            pw = True
            s += line
            continue

        if pw is False:
            s += line
            continue

        if re.search('\+\+\+\-exit\-\+\+\+ (\d+)', line):
            s += line
            exitcode = 1
            break

        sys.stdout.write(line)

    r.close()

    if pw is False:
        exitcode = 2

    if exitcode:
        sys.stderr.write(s)
        sys.stderr.write('exit ' + str(exitcode) + '\n')

    sys.exit(exitcode)

if __name__ == '__main__':
    main()